src/6_branch.js

```diff
@@ -67,16 +67,23 @@ var wrap = function(parameters, func, init) {
 			 * @type {function(?Error,?): undefined}
 			 */
 			var done = function(err, data) {
-				if (err && parameters === callback_params.NO_CALLBACK) {
+				try {
+					if (err && parameters === callback_params.NO_CALLBACK) {
+						throw err;
+					}
+					else if (parameters === callback_params.CALLBACK_ERR) {
+						callback(err);
+					}
+					else if (parameters === callback_params.CALLBACK_ERR_DATA) {
+						callback(err, data);
+					}
+				} catch (err) {
+					// the callback functions may or may not throw an error...
 					throw err;
```

Comment:
the point of `throw` is to yell and halt the execution because presumably from that point on further execution will be pointless; but here we don't want to halt.
Therefore can we get away with simply `console.error(error message here)`? This try-throw-finally block smells.